array-checker.js

```diff
@@ -0,0 +1,15 @@
+function uniqueCalc(arr1, arr2) {
+  for (var i = 0; i < arr1.length; i++) {
+    for (var j = 0; j < arr2.length; j++) {
+      if (arr1[i] === arr2[j]) {
+        arr1 = arr1.splice(i, 1);
```

Comment:
This is very bad here for 2 reasons:

1. Time complexity here is very high O((n ^ 2) * m), as `splice` internally will loop over the array to re-organize its elements. It's better here to define a new array and use it for adding unique items, which will keep the time complexity at O(n * m) (this isn't great, but it's better).

2. The original `arr1` array is now being changed a lot, while you are actively looping it, which could very easily cause bugs.